verification/curator-service/ui/src/components/common-form-fields/Source.tsx

```diff
@@ -185,8 +185,24 @@ export function SourcesAutocomplete(
                 }}
                 onChange={(
                     _: any,
-                    newValue: CaseReferenceForm | null,
+                    newValue: CaseReferenceForm | string | null,
                 ): void => {
+                    // newValue is a string if the user typed a URL and did not
+                    // select a dropdown value.
+                    if (typeof newValue === 'string') {
+                        const existingOption = options.find(
+                            (option) => option.sourceUrl === newValue,
+                        );
+                        newValue = existingOption ?? {
+                            inputValue: newValue,
+                            sourceUrl: newValue,
+                            sourceId: '',
+                            sourceName: '',
+                            additionalSources: ([] as unknown) as [
```

Comment:
why is the `[] as unknown` needed?

Reply:
This is the error message when it is removed: `Conversion of type '[]' to type '[{ sourceUrl: string; }]' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.`